From 7199e9650cf0029f63c44b1244a415a977736f59 Mon Sep 17 00:00:00 2001
From: Mirjam Friesen <mf@travelping.com>
Date: Wed, 18 Dec 2013 15:01:12 +0100
Subject: [PATCH] fix/boss_compiler

---
 src/lager_transform.erl | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/lager_transform.erl b/src/lager_transform.erl
index f610684..2ffdb49 100644
--- a/src/lager_transform.erl
+++ b/src/lager_transform.erl
@@ -200,6 +200,7 @@ transform_statement(Stmt) when is_list(Stmt) ->
 transform_statement(Stmt) ->
     Stmt.
 
+make_varname(Prefix, {Line, _}) -> make_varname(Prefix, Line);
 make_varname(Prefix, Line) ->
     list_to_atom(Prefix ++ atom_to_list(get(module)) ++ integer_to_list(Line)).
 
-- 
1.8.3.1

