Index: git/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/Makefile.am	2012-08-07 18:10:30.168101734 +0200
@@ -0,0 +1,20 @@
+#
+#
+#
+ACLOCAL_AMFLAGS = -I m4
+
+EXTRA_DIST = README
+SUBDIRS = include lib/common lib/config lib/rofl socktest
+
+doc:
+	@doxygen doxyfile.conf
+
+indent:
+	@if [ -z $$(which uncrustify) ]; then \
+		echo "Please install uncrustify (e.g. `apt-get install uncrustify`)"; \
+	else \
+		uncrustify -c unc_indent.cfg --replace include/*.h; \
+		for dir in $(SUBDIRS); do \
+			make -C $$dir $@; \
+		done; \
+	fi
Index: git/configure.ac
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/configure.ac	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,70 @@
+#                                               -*- Autoconf -*-
+# Process this file with autoconf to produce a configure script.
+
+AC_PREREQ([2.68])
+AC_INIT([rofl],[0.1],[bugs@rofl.com])
+AC_CONFIG_SRCDIR([lib/rofl/clinuxport.cc])
+AC_CONFIG_AUX_DIR(auxdir)
+AM_INIT_AUTOMAKE([foreign silent-rules subdir-objects -Wcross])
+AC_CONFIG_HEADERS([config.h])
+AC_CONFIG_MACRO_DIR([m4])
+
+dnl This mechanism allows one to enable debug compilations...
+AC_MSG_CHECKING([if debugging is wanted])
+AC_ARG_ENABLE([debug],
+  [AS_HELP_STRING([--disable-debug],
+                  [disable debugging])],
+  [], [ enable_debug=yes ])
+if test "$enable_debug" = "yes"; then
+   AC_MSG_RESULT(yes)
+   CXXFLAGS="$CXXFLAGS -DDEBUG"
+else
+   AC_MSG_RESULT(no)
+   CXXFLAGS="$CXXFLAGS -DNDEBUG"
+fi
+AM_CONDITIONAL(ENABLE_DEBUG, test "$enable_debug" = "yes")
+
+ORAN_FLAGS=-DORAN
+AC_SUBST([ORAN_FLAGS])
+
+# Checks for programs.
+AC_PROG_CXX
+AC_PROG_CC
+AM_PROG_CC_C_O
+AC_PROG_INSTALL
+AC_PROG_MAKE_SET
+LT_INIT
+
+AC_CHECK_HEADERS([libcli.h],, [AC_MSG_ERROR([cli library include file missing])])
+AC_CHECK_LIB([cli], [cli_command_name],, [AC_MSG_ERROR([cli library missing])])
+
+AC_CHECK_LIB([pthread], [pthread_mutex_init],, [AC_MSG_ERROR([pthread library missing])])
+AC_CHECK_LIB([rt], [clock_gettime])
+
+# Checks for header files.
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h inttypes.h limits.h netinet/in.h stdint.h stdlib.h string.h strings.h sys/ioctl.h sys/socket.h sys/time.h syslog.h unistd.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_HEADER_STDBOOL
+AC_C_INLINE
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT16_T
+AC_TYPE_UINT32_T
+AC_TYPE_UINT64_T
+AC_TYPE_UINT8_T
+
+# Checks for library functions.
+AC_FUNC_FORK
+AC_FUNC_MMAP
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([bzero clock_gettime dup2 getpagesize memmove memset munmap select socket strerror strncasecmp])
+
+AC_CONFIG_FILES([
+  Makefile
+  include/Makefile
+  lib/common/Makefile
+  lib/config/Makefile
+  lib/rofl/Makefile
+  socktest/Makefile])
+AC_OUTPUT
Index: git/lib/common/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/lib/common/Makefile.am	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,25 @@
+#
+#
+#
+
+noinst_LTLIBRARIES = librofcommon-noerror.la
+librofcommon_noerror_la_SOURCES = cgraph.cc cofmatch.cc
+librofcommon_noerror_la_CXXFLAGS = $(ORAN_FLAGS) -Wno-error -I$(top_srcdir)/include
+
+lib_LTLIBRARIES = librofcommon.la
+librofcommon_la_SOURCES = \
+	caddress.cc cclock.cc cctlelem.cc cerror.cc cfibentry.cc cflowentry.cc \
+	cforward.cc cfsm.cc cfspentry.cc cfsptable.cc cftentry.cc cfttable.cc \
+	cfwdelem.cc cfwdentry.cc cfwdtable.cc cgroupentry.cc cgtentry.cc \
+	cgttable.cc ciosrv.cc clldplist.cc clldptlv.cc cmacaddr.cc cmemory.cc \
+	cmempool.cc cofaclist.cc cofaction.cc cofactionset.cc cofbase.cc \
+	cofbclist.cc cofbucket.cc cofctrl.cc cofdpath.cc cofinlist.cc cofinst.cc \
+	coflist.cc cofpacket.cc cofport.cc cofportlist.cc cofrpc.cc \
+	coxmatch.cc coxmlist.cc cpacket.cc cpipe.cc cport.cc cpppoelist.cc \
+	cpppoetlv.cc crandom.cc croflexp.cc csocket.cc csyslog.cc cutbase.cc \
+	farpv4frame.cc fetherframe.cc fframe.cc ficmpv4frame.cc fipv4frame.cc \
+	flldpframe.cc fmplsframe.cc fpppframe.cc fpppoeframe.cc ftcpframe.cc \
+	fudpframe.cc fvlanframe.cc
+librofcommon_la_LIBADD = librofcommon-noerror.la
+librofcommon_la_LDFLAGS = -version-info 0:1:0
+librofcommon_la_CXXFLAGS = $(ORAN_FLAGS) -I$(top_srcdir)/include
\ No newline at end of file
Index: git/lib/common/makefile
===================================================================
--- git.orig/lib/common/makefile	2012-08-07 18:10:29.792095953 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,63 +0,0 @@
-#
-#
-#
-BASEDIR := ../..
-
-include $(BASEDIR)/makefile.inc
-
-#TARGET := librofl.so.0.1.0
-TARGET := librofcommon.a
-STATIC_TARGET := librofcommon.a
-INCDIR := -I $(BASEDIR)/include -I $(BASEDIR)/../libcli
-OBJDIR := build
-INCS := $(wildcard *.h)
-SRCS := $(wildcard *.cc)
-OBJS := $(patsubst %.cc,$(OBJDIR)/%.o,$(SRCS))
-
-LIBS := -L $(BASEDIR)/lib -lrt
-
-CFLAGS+= $(INCDIR)
-
-
-all: $(TARGET) 
-#all: $(TARGET) subdirs
-
-subdirs: $(SUBDIRS)
-			$(MAKE) -C $?
-
-
-.depend: $(SRCS)
-			$(CC) -MM $(SRCS) $(INCS) $(INCDIR) > .depend
-
-
-$(OBJDIR)/cgraph.o : cgraph.cc
-	$(CC) $(CFLAGS) -Wno-error -c $< -o $@
-
-$(OBJDIR)/cofmatch.o : cofmatch.cc
-	$(CC) $(CFLAGS) -Wno-error -c $< -o $@
-
-$(OBJDIR)/%.o : %.cc
-			$(CC) $(CFLAGS) -c $< -o $@
-
-$(TARGET): $(OBJS) .depend
-			$(CROSS_COMPILE)ar rcs $(STATIC_TARGET) $(OBJS)
-			#$(CC) -shared -Wl,-soname,librofl.so.0 -o librofl.so.0.1.0 $(OBJS) ${LIBS}
-			#ln -sf librofl.so.0.1.0 librofl.so.0.1
-			#ln -sf librofl.so.0.1 librofl.so.0
-			#ln -sf librofl.so.0 librofl.so
-
-$(OBJS): | $(OBJDIR)
-    
-$(OBJDIR):
-		mkdir $(OBJDIR)
-
-.PHONY : clean rclean
-clean :
-			rm -f $(STATIC_TARGET)
-			rm -f $(OBJS)
-
-rclean : clean
-			rm -f .depend $(TARGET)
-			rm -f *~
-
-include .depend
Index: git/lib/config/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/lib/config/Makefile.am	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,8 @@
+#
+#
+#
+
+lib_LTLIBRARIES = librofconfig.la
+librofconfig_la_SOURCES = ccli.cc cconfigfwdelem.cc cconfigport.cc
+librofconfig_la_LDFLAGS = -version-info 0:1:0
+librofconfig_la_CXXFLAGS = $(ORAN_FLAGS) -I$(top_srcdir)/include
\ No newline at end of file
Index: git/lib/config/makefile
===================================================================
--- git.orig/lib/config/makefile	2012-08-07 18:10:29.792095953 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,68 +0,0 @@
-#
-#
-#
-BASEDIR := ../..
-
-include $(BASEDIR)/makefile.inc
-
-#TARGET := librofl.so.0.1.0
-TARGET := librofconfig.a
-STATIC_TARGET := librofconfig.a
-INCDIR := -I $(BASEDIR)/include -I $(BASEDIR)/../libcli
-OBJDIR := build
-INCS := $(wildcard *.h)
-SRCS := $(wildcard *.cc)
-OBJS := $(patsubst %.cc,$(OBJDIR)/%.o,$(SRCS))
-ADDITIONAL_OBJ := $(BASEDIR)/../libcli/libcli.o
-
-LIBS := -L $(BASEDIR)/lib -lrt -llibcrypt
-
-CFLAGS+= $(INCDIR)
-
-
-all: $(TARGET) 
-#all: $(TARGET) subdirs
-
-subdirs: $(SUBDIRS)
-			$(MAKE) -C $?
-
-
-.depend: $(SRCS)
-			$(CC) -MM $(SRCS) $(INCS) $(INCDIR) > .depend
-
-
-$(OBJDIR)/cgraph.o : cgraph.cc
-	$(CC) $(CFLAGS) -Wno-error -c $< -o $@
-
-$(OBJDIR)/cofmatch.o : cofmatch.cc
-	$(CC) $(CFLAGS) -Wno-error -c $< -o $@
-
-$(OBJDIR)/%.o : %.cc
-			$(CC) $(CFLAGS) -c $< -o $@
-
-
-$(ADDITIONAL_OBJ):
-	@if [ ! -f $(ADDITIONAL_OBJ) ]; then echo "missing $(ADDITIONAL_OBJ)";exit -1; fi; 
-
-$(TARGET): $(OBJS) .depend $(ADDITIONAL_OBJ)
-			$(CROSS_COMPILE)ar rcs $(STATIC_TARGET) $(OBJS) $(ADDITIONAL_OBJ)
-			#$(CC) -shared -Wl,-soname,librofl.so.0 -o librofl.so.0.1.0 $(OBJS) ${LIBS}
-			#ln -sf librofl.so.0.1.0 librofl.so.0.1
-			#ln -sf librofl.so.0.1 librofl.so.0
-			#ln -sf librofl.so.0 librofl.so
-
-$(OBJS): | $(OBJDIR)
-    
-$(OBJDIR):
-		mkdir $(OBJDIR)
-
-.PHONY : clean rclean
-clean :
-			rm -f $(STATIC_TARGET)
-			rm -f $(OBJS)
-
-rclean : clean
-			rm -f .depend $(TARGET)
-			rm -f *~
-
-include .depend
Index: git/lib/makefile
===================================================================
--- git.orig/lib/makefile	2012-08-07 18:10:29.796096014 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,25 +0,0 @@
-#
-#
-#
-
-BASEDIR := ..
-SUBDIRS := common rofl config
-
-all: $(SUBDIRS)
- 
-.PHONY: $(SUBDIRS)
-
-$(SUBDIRS):
-			$(MAKE) -C $@
-
-.PHONY : clean rclean
-clean :
-	@for dir in $(SUBDIRS); do \
-		make -C $$dir clean; \
-	done;
-
-rclean : clean
-	@for dir in $(SUBDIRS); do \
-		make -C $$dir rclean; \
-	done;
-
Index: git/lib/rofl/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/lib/rofl/Makefile.am	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,9 @@
+#
+#
+#
+lib_LTLIBRARIES = librofl.la
+librofl_la_SOURCES = \
+	cethport.cc clinuxport.cc cmmapport.cc cnetmapport.cc cofipv4port.cc \
+	cpkbuf.cc crib.cc ctapport.cc cunixenv.cc cunixport.cc
+librofl_la_LDFLAGS = -version-info 0:1:0
+librofl_la_CXXFLAGS = $(ORAN_FLAGS) -I$(top_srcdir)/include
\ No newline at end of file
Index: git/lib/rofl/makefile
===================================================================
--- git.orig/lib/rofl/makefile	2012-08-07 18:10:29.796096014 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,56 +0,0 @@
-#
-#
-#
-BASEDIR := ../..
-
-include $(BASEDIR)/makefile.inc
-
-#TARGET := librofl.so.0.1.0
-TARGET := librofl.a
-STATIC_TARGET := librofl.a
-INCDIR := -I $(BASEDIR)/include
-OBJDIR := build
-INCS := $(wildcard *.h)
-SRCS := $(wildcard *.cc)
-OBJS := $(patsubst %.cc,$(OBJDIR)/%.o,$(SRCS))
-
-LIBS := -L $(BASEDIR)/lib
-
-CFLAGS+= $(INCDIR)
-
-all: $(TARGET) 
-#all: $(TARGET) subdirs
-
-subdirs: $(SUBDIRS)
-			$(MAKE) -C $?
-
-
-.depend: $(SRCS)
-			$(CC) -MM $(SRCS) $(INCS) $(INCDIR) > .depend
-     
-$(OBJDIR)/%.o : %.cc
-			$(CC) $(CFLAGS) -c $< -o $@
-     
-
-$(TARGET): $(OBJS) .depend
-			$(CROSS_COMPILE)ar rcs $(STATIC_TARGET) $(OBJS)
-			#$(CC) -shared -Wl,-soname,librofl.so.0 -o librofl.so.0.1.0 $(OBJS) ${LIBS}
-			#ln -sf librofl.so.0.1.0 librofl.so.0.1
-			#ln -sf librofl.so.0.1 librofl.so.0
-			#ln -sf librofl.so.0 librofl.so
-
-$(OBJS): | $(OBJDIR)
-    
-$(OBJDIR):
-		mkdir $(OBJDIR)
-
-.PHONY : clean rclean
-clean :
-			rm -f $(STATIC_TARGET)
-			rm -f $(OBJS)
-
-rclean : clean
-			rm -f .depend $(TARGET)
-			rm -f *~
-
-include .depend
Index: git/m4/.gitignore
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/m4/.gitignore	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,6 @@
+libtool.m4
+lt~obsolete.m4
+ltoptions.m4
+ltsugar.m4
+ltversion.m4
+
Index: git/makefile
===================================================================
--- git.orig/makefile	2012-08-07 18:10:29.796096014 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,39 +0,0 @@
-#
-#
-#
-
-include makefile.inc
-
-SHELL = /bin/bash
-
-ifneq ($(CROSS_COMPILE),)
-SUBDIRS = lib 
-else
-SUBDIRS = lib socktest
-endif
-
-.PHONY: rclean clean $(SUBDIRS)
-
-all: $(SUBDIRS)
-
-$(SUBDIRS):
-	$(MAKE) -C $@
-
-clean:
-	@for dir in $(SUBDIRS); do \
-		make -C $$dir $@; \
-	done; 
-
-rclean: clean
-	@for dir in $(SUBDIRS); do \
-		make -C $$dir $@; \
-	done;
-	rm -rf ./include/*~
-	rm -rf ./doc Doxyfile doxygen.file
-	rm -f *~
-
-distclean: rclean
-
-doc:
-	@doxygen doxyfile.conf
-
Index: git/makefile.inc
===================================================================
--- git.orig/makefile.inc	2012-08-07 18:10:29.796096014 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,30 +0,0 @@
-CC=${CROSS_COMPILE}g++
-
-CFLAGS=-Wall -Werror -pthread
-
-# enable debugging by default
-DEBUG ?= 1
-ifeq ($(DEBUG),1)
-CFLAGS += -g -DDEBUG
-else
-CFLAGS += -DNDEBUG -g -O2
-endif
-
-# enable deep debugging during development (e.g. flush each writelog call)
-CFLAGS += -DDEBUG_DEVEL
-
-# profiling
-CFLAGS += -pg
-
-# enable memory pool
-# CFLAGS += -DMEMPOOL
-
-# enable ORAN support
-CFLAGS += -DORAN
-
-# netmap support and includes
-# CFLAGS += -DNETMAP -I${HOME}/src/netmap-linux/include
-
-# enable performance testing for L2 forwarding
-# CFLAGS += -DSWITCHINGTEST
-
Index: git/socktest/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/socktest/Makefile.am	2012-08-07 18:10:30.172101796 +0200
@@ -0,0 +1,13 @@
+#
+#
+#
+
+sbin_PROGRAMS = socktest
+socktest_SOURCES = chub.cc csocktest.cc socktest.cc
+socktest_CXXFLAGS = $(ORAN_FLAGS) \
+		    -I$(srcdir)/include \
+		    -I$(top_srcdir)/include \
+		    -I$(top_srcdir)/include/common \
+		    -I$(top_srcdir)/include/config \
+		    -I$(top_srcdir)/include/rofl
+socktest_LDADD = $(top_builddir)/lib/rofl/librofl.la $(top_builddir)/lib/common/librofcommon.la
Index: git/socktest/makefile
===================================================================
--- git.orig/socktest/makefile	2012-08-07 18:10:29.796096014 +0200
+++ /dev/null	1970-01-01 00:00:00.000000000 +0000
@@ -1,45 +0,0 @@
-#
-#
-#
-
-TARGET := socktest
-BASEDIR := ..
-INCDIR := -I $(BASEDIR)/include -I.
-OBJDIR := .
-SRCDIR := src
-INCS := $(wildcard *.h)
-SRCS := $(wildcard *.cc)
-OBJS := $(patsubst %.cc,%.o,$(SRCS))
-# LIBS := -lstdc++ -ldl -L ../lib/datapath -lrofl -L ../lib/common -l rofcommon
-LIBS := -lstdc++ -ldl -lrt -lpthread ../lib/rofl/librofl.a ../lib/common/librofcommon.a -lrt
-#LIBS := -lstdc++ -ldl ../lib/common/librofcommon.a ../lib/rofl/librofl.a 
-
-CC      = $(CROSS_COMPILE)g++
-CFLAGS  = -Wall -g -pg $(INCDIR) -fPIC -DORAN
-
-all: $(TARGET)
-
-.depend: $(SRCS)
-			$(CC) -MM $(SRCS) $(INCS) $(INCDIR) > .depend
-     
-$(OBJDIR)/%.o : %.cc
-			$(CC) $(CFLAGS) -c $< -o $@
-     
-
-$(TARGET): $(OBJS) .depend
-			$(CC) $(CFLAGS) -o $@ $(OBJS) ${LIBS}
-
-$(OBJS): | $(OBJDIR)
-    
-$(OBJDIR):
-		mkdir $(OBJDIR)
-
-.PHONY : clean rclean
-clean :
-			rm -f $(OBJS)
-
-rclean : clean
-			rm -f .depend $(TARGET)
-			rm -f *~
-			
-include .depend
Index: git/include/Makefile.am
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ git/include/Makefile.am	2012-08-07 18:10:59.488552543 +0200
@@ -0,0 +1,25 @@
+#
+#
+#
+
+nobase_pkginclude_HEADERS = \
+	common/caddress.h common/cclock.h common/cctlelem.h common/cerror.h common/cfibentry.h \
+	common/cflowentry.h common/cforward.h common/cfsm.h common/cfspentry.h common/cfsptable.h \
+	common/cftentry.h common/cfttable.h common/cfwdelem.h common/cfwdentry.h common/cfwdtable.h \
+	common/cgraph.h common/cgroupentry.h common/cgtentry.h common/cgttable.h common/ciosrv.h \
+	common/clldplist.h common/clldptlv.h common/cmacaddr.h common/cmemory.h common/cmempool.h \
+	common/cofaclist.h common/cofaction.h common/cofactionset.h common/cofbase.h \
+	common/cofbclist.h common/cofbucket.h common/cofctrl.h common/cofdpath.h common/cofinlist.h \
+	common/cofinst.h common/coflist.h common/cofmatch.h common/cofpacket.h common/cofport.h \
+	common/cofportlist.h common/cofrpc.h common/coxmatch.h common/coxmlist.h common/cpacket.h \
+	common/cpipe.h common/cport.h common/cpppoelist.h common/cpppoetlv.h common/crandom.h \
+	common/croflexp.h common/csocket.h common/csyslog.h common/cutbase.h common/cvastring.h \
+	common/endian_conversion.h common/farpv4frame.h common/fetherframe.h common/fframe.h \
+	common/ficmpv4frame.h common/fipv4frame.h common/flldpframe.h common/fmplsframe.h \
+	common/fpppframe.h common/fpppoeframe.h common/ftcpframe.h common/fudpframe.h \
+	common/fvlanframe.h common/lldp.h common/openflow11.h \
+	common/openflow12.h common/openflow_rofl.h common/rofcommon.h common/thread_helper.h \
+	config/ccli.h config/cconfigfwdelem.h config/cconfigport.h \
+	rofl/cethport.h rofl/clinuxport.h rofl/cmmapport.h rofl/cnetmapport.h \
+	rofl/cofipv4port.h rofl/cpkbuf.h rofl/crib.h rofl/ctapport.h rofl/cunixenv.h \
+	rofl/cunixport.h
\ No newline at end of file
