From 2eab8995e0afabd7914db0df61070811ce361abc Mon Sep 17 00:00:00 2001
From: Tobias Hintze <thintze+git@tpip.net>
Date: Fri, 8 May 2015 13:37:03 +0200
Subject: [PATCH] adapt Makefile to caros systemd

---
 c_src/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/c_src/Makefile b/c_src/Makefile
index a900bf6..def8ff4 100644
--- a/c_src/Makefile
+++ b/c_src/Makefile
@@ -3,10 +3,10 @@ CC ?= gcc
 RM ?= rm
 CFLAGS ?= -O2
 
-override CFLAGS += -std=gnu99  -Wall -fpic -I. $(shell erl -noinput -eval 'io:format("-I~s/erts-~s/include", [code:root_dir(), erlang:system_info(version)]), halt(0).') $(shell pkg-config --cflags libsystemd-journal)
+override CFLAGS += -std=gnu99  -Wall -fpic -I. $(shell erl -noinput -eval 'io:format("-I~s/erts-~s/include", [code:root_dir(), erlang:system_info(version)]), halt(0).') $(shell pkg-config --cflags libsystemd)
 
 override LDFLAGS += -fpic
-LDLIBS = $(shell pkg-config --libs libsystemd-journal)
+LDLIBS = $(shell pkg-config --libs libsystemd)
 
 all : $(PRIVDIR)/journald_api.so
 
-- 
1.9.1

