Index: git/rel/overlay/etc/local.ini
===================================================================
--- git.orig/rel/overlay/etc/local.ini	2014-07-08 14:33:57.991775618 +0200
+++ git/rel/overlay/etc/local.ini	2014-07-08 14:34:56.227777178 +0200
@@ -28,6 +28,9 @@
 [httpd_global_handlers]
 ;_google = {couch_httpd_proxy, handle_proxy_req, <<"http://www.google.com">>}
 
+[native_query_servers]
+erlang = {couch_native_process, start_link, []}
+
 [couch_httpd_auth]
 ; If you set this to true, you should also uncomment the WWW-Authenticate line
 ; above. If you don't configure a WWW-Authenticate header, CouchDB will send
