diff --git a/couchjs/SConstruct b/couchjs/SConstruct
index 2b57cb4..2a8bea0 100644
--- a/couchjs/SConstruct
+++ b/couchjs/SConstruct
@@ -12,5 +12,9 @@
 # License for the specific language governing permissions and limitations under
 # the License.
 
+import os
+
+env = Environment(ENV = os.environ)
+
 SConscript(['c_src/SConscript'], variant_dir='build')
 SConscript(['js/SConscript'])
diff --git a/couchjs/c_src/SConscript b/couchjs/c_src/SConscript
index 3f6a0cc..b19e22d 100644
--- a/couchjs/c_src/SConscript
+++ b/couchjs/c_src/SConscript
@@ -22,8 +22,8 @@ def require_lib(name):
 def runcmd(cmd):
     return commands.getstatusoutput(cmd)
 
-env = Environment(CC="c++", CCFLAGS='-g -O2 -DXP_UNIX',
-                        CPPPATH=os.getenv("CPPPATH"))
+env = Environment(ENV = os.environ)
+env['CCFLAGS'] = '-DXP_UNIX'
 
 if os.uname()[0] == 'SunOS':
    env['CC'] = '/usr/sfw/bin/gcc'
diff --git a/couchjs/scons/scons-local-2.0.1/SCons/Environment.py b/couchjs/scons/scons-local-2.0.1/SCons/Environment.py
index d5206fa..28f0672 100644
--- a/couchjs/scons/scons-local-2.0.1/SCons/Environment.py
+++ b/couchjs/scons/scons-local-2.0.1/SCons/Environment.py
@@ -1392,6 +1392,13 @@ class Base(SubstitutionEnvironment):
         if not SCons.Util.is_List(progs):
             progs = [ progs ]
         for prog in progs:
+            # prefer cross version of program, if it exists
+            cross_prog = os.environ.get('CROSS_COMPILE') + prog
+            #print "Looking for %s" % cross_prog
+            path = self.WhereIs(cross_prog)
+            if path: return cross_prog
+
+            #print "Looking for %s" % prog
             path = self.WhereIs(prog)
             if path: return prog
         return None
