diff -up hostapd-1.1/src/drivers/driver_capwap_wtp.c.orig hostapd-1.1/src/drivers/driver_capwap_wtp.c
--- hostapd-1.1/src/drivers/driver_capwap_wtp.c.orig	2013-09-20 14:44:50.323182383 +0200
+++ hostapd-1.1/src/drivers/driver_capwap_wtp.c	2013-09-20 14:53:26.775175464 +0200
@@ -6670,6 +6670,20 @@ static int i802_sta_deauth(void *priv, c
 	struct i802_bss *bss = priv;
 	struct ieee80211_mgmt mgmt;
 
+	/* fake deauth from station for capwap */
+	memset(&mgmt, 0, sizeof(mgmt));
+	mgmt.frame_control = IEEE80211_FC(WLAN_FC_TYPE_MGMT,
+					  WLAN_FC_STYPE_DEAUTH);
+	/* sa/da reversed */
+	memcpy(mgmt.da, own_addr, ETH_ALEN);
+	memcpy(mgmt.sa, addr, ETH_ALEN);
+	memcpy(mgmt.bssid, own_addr, ETH_ALEN);
+	mgmt.u.deauth.reason_code = host_to_le16(reason);
+
+	ipc_send_80211_to_ac(generic_wtp_info.fd_ipc, (u8 *) &mgmt,
+			     IEEE80211_HDRLEN + sizeof(mgmt.u.deauth));
+	/* -------------*/
+
 	memset(&mgmt, 0, sizeof(mgmt));
 	mgmt.frame_control = IEEE80211_FC(WLAN_FC_TYPE_MGMT,
 					  WLAN_FC_STYPE_DEAUTH);
