From: Tobias Hintze <thintze+git@tpip.net>
Date: Thu Feb 11 12:38:20 CET 2016
Subject: [PATCH] reduce verbosity to DEBUG

diff -Naurw opkg-0.3.0/libopkg/pkg_hash.c opkg-0.3.0-new/libopkg/pkg_hash.c
--- opkg-0.3.0-orig/libopkg/pkg_hash.c	2015-05-28 19:29:19.000000000 +0200
+++ opkg-0.3.0-new/libopkg/pkg_hash.c	2016-02-11 12:37:05.409604266 +0100
@@ -373,7 +373,8 @@
              * force_depends is set.
              */
             if (pkg_breaks_reverse_dep(maybe) && !opkg_config->force_depends) {
-                opkg_msg(NOTICE,
+                /* this is actually a verbosity > NOTICE. */
+                opkg_msg(DEBUG,
                          "Not selecting %s %s as installing it would break "
                          "existing dependencies.\n",
                          maybe->name, maybe->version);
