From 19f40712c1c786a3530119d462d1c4350c180e9c Mon Sep 17 00:00:00 2001
From: Tobias Hintze <thintze+git@tpip.net>
Date: Thu, 4 Feb 2016 16:13:29 +0100
Subject: [PATCH] reduce verbosity

---
 libopkg/pkg_hash.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/libopkg/pkg_hash.c b/libopkg/pkg_hash.c
index 91e413c..7e3a276 100644
--- a/libopkg/pkg_hash.c
+++ b/libopkg/pkg_hash.c
@@ -369,8 +369,12 @@ pkg_hash_fetch_best_installation_candidate(abstract_pkg_t *apkg,
 	       */
 	      if (pkg_breaks_reverse_dep(maybe) &&
 		      !conf->force_depends) {
-		  opkg_msg(NOTICE, "Not selecting %s %s as installing it would break existing dependencies.\n",
+		  /*
+		   * this is actually a verbosity > NOTICE.
+		   *
+			  opkg_msg(NOTICE, "Not selecting %s %s as installing it would break existing dependencies.\n",
 			   maybe->name, maybe->version);
+		   */
 		  continue;
 	      }
 
-- 
1.9.1

