Index: bind-9.9.5/conf/db.255
===================================================================
--- /dev/null
+++ bind-9.9.5/conf/db.255
@@ -0,0 +1,12 @@
+;
+; BIND reverse data file for broadcast zone
+;
+$TTL   604800
+@  IN  SOA localhost. root.localhost. (
+                 1     ; Serial
+            604800     ; Refresh
+             86400     ; Retry
+           2419200     ; Expire
+            604800 )   ; Negative Cache TTL
+;
+@  IN  NS  localhost.
