Index: qemu-2.1.0-rc0-orig/block/vmdk.c
===================================================================
--- qemu-2.1.0-rc0-orig.orig/block/vmdk.c	2014-07-08 17:47:14.504086163 +0200
+++ qemu-2.1.0-rc0-orig/block/vmdk.c	2014-07-08 17:45:51.720083947 +0200
@@ -645,7 +645,9 @@
         error_set(errp, QERR_UNKNOWN_BLOCK_FORMAT_FEATURE,
                   bs->device_name, "vmdk", buf);
         return -ENOTSUP;
-    } else if (le32_to_cpu(header.version) == 3 && (flags & BDRV_O_RDWR)) {
+    } else if (le32_to_cpu(header.version) == 3 &&
+	       (flags & BDRV_O_RDWR) &&
+	       !(le32_to_cpu(header.flags) & VMDK4_FLAG_COMPRESS)) {
         /* VMware KB 2064959 explains that version 3 added support for
          * persistent changed block tracking (CBT), and backup software can
          * read it as version=1 if it doesn't care about the changed area
@@ -1562,7 +1564,7 @@
     }
     magic = cpu_to_be32(VMDK4_MAGIC);
     memset(&header, 0, sizeof(header));
-    header.version = zeroed_grain ? 2 : 1;
+    header.version = zeroed_grain ? 2 : (compress ? 3 : 1);
     header.flags = VMDK4_FLAG_RGD | VMDK4_FLAG_NL_DETECT
                    | (compress ? VMDK4_FLAG_COMPRESS | VMDK4_FLAG_MARKER : 0)
                    | (zeroed_grain ? VMDK4_FLAG_ZERO_GRAIN : 0);
