diff --git a/tetrapak/config.ini b/tetrapak/config.ini
index fd5245e..ded2517 100644
--- a/tetrapak/config.ini
+++ b/tetrapak/config.ini
@@ -5,7 +5,7 @@ erlc_options = [{parse_transform, lager_transform}]
 [package]
 include_src = false
 deb.section = "net"
-exclude = "\\.gitignore|TODO\\.rst|example\\.config|sample"
+exclude = "\\.gitignore|TODO\\.rst|example\\.config|sample|rebar.*"
 
 [xref]
 ignore_undef = [ { erlang, timestamp, 0 } ]
