From 27f160c1372cf02b4b233f7e296b00ec978752e7 Mon Sep 17 00:00:00 2001
From: Tino Breddin <tb@travelping.com>
Date: Fri, 21 Jun 2013 08:08:56 +0200
Subject: [PATCH] Make configure cross-compile compatible

---
 configure | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure b/configure
index da01c19..c64fa0d 100755
--- a/configure
+++ b/configure
@@ -2,6 +2,7 @@
 # This is not an autoconf generated configure
 #
 INCLUDE=${1:-"$PWD/include"}
+SYSROOT=$1
 
 # Make a temp directory in build tree.
 TMPDIR=$(mktemp -d config.XXXXXX)
@@ -166,7 +167,7 @@ check_ipt_lib_dir()
 		return
 	fi
 
-	for dir in /lib /usr/lib /usr/local/lib
+    	for dir in $SYSROOT/lib $SYSROOT/usr/lib $SYSROOT/usr/local/lib
 	do
 		for file in $dir/{xtables,iptables}/lib*t_*so ; do
 			if [ -f $file ]; then
-- 
1.8.3.1


