Upstream-Status: Inappropriate [configuration]

Index: lrzsz-0.12.20/acinclude.m4
===================================================================
--- lrzsz-0.12.20.orig/acinclude.m4	1998-12-27 23:08:59.000000000 +0100
+++ lrzsz-0.12.20/acinclude.m4	2013-03-28 12:22:48.200425128 +0100
@@ -1,12 +1,11 @@
 dnl AC_REPLACE_GNU_GETOPT
-AC_DEFUN(AC_REPLACE_GNU_GETOPT,
-[AC_CHECK_FUNC(getopt_long, , [LIBOBJS="$LIBOBJS getopt1.o getopt.o"])
-AC_SUBST(LIBOBJS)dnl
-])
+AC_DEFUN([AC_REPLACE_GNU_GETOPT],
+[AC_CHECK_FUNC(getopt_long, , [AC_LIBOBJ([getopt1])
+AC_LIBOBJ([getopt])dnl])])
 
 dnl
 dnl taken from taylor uucp
-AC_DEFUN(LRZSZ_ERRNO_DECL,[
+AC_DEFUN([LRZSZ_ERRNO_DECL],[
 AC_MSG_CHECKING(for errno declaration)
 AC_CACHE_VAL(lrzsz_cv_decl_errno,
 [AC_TRY_COMPILE([#include <errno.h>], [int i = errno; errno = 1;],
@@ -23,7 +22,7 @@
 ])
 
 dnl LRZSZ_TYPE_SPEED_T
-AC_DEFUN(LRZSZ_TYPE_SPEED_T,[
+AC_DEFUN([LRZSZ_TYPE_SPEED_T],[
 AC_REQUIRE([AC_HEADER_STDC])dnl
 AC_REQUIRE([LRZSZ_HEADERS_TERM_IO])dnl
 AC_MSG_CHECKING(for speed_t)
@@ -61,7 +60,7 @@
 fi
 ])
 
-AC_DEFUN(lrzsz_HEADER_SYS_SELECT,
+AC_DEFUN([lrzsz_HEADER_SYS_SELECT],
 [AC_CACHE_CHECK([whether sys/time.h and sys/select.h may both be included],
   lrzsz_cv_header_sys_select,
 [AC_TRY_COMPILE([#include <sys/types.h>
Index: lrzsz-0.12.20/configure.in
===================================================================
--- lrzsz-0.12.20.orig/configure.in	2013-03-28 12:25:35.000000000 +0100
+++ lrzsz-0.12.20/configure.in	2013-03-28 12:25:56.156425003 +0100
@@ -132,6 +132,7 @@
 AC_CHECK_HEADERS(termios.h sys/termios.h termio.h sys/termio.h sgtty.h)
 AC_CHECK_HEADERS(sys/mman.h utime.h syslog.h sys/syslog.h sys/param.h)
 AC_CHECK_HEADERS(sys/select.h strings.h arpa/inet.h)
+AC_CHECK_HEADERS(locale.h)
 
 dnl Checks for typedefs, structures, and compiler characteristics.
 AC_TYPE_SIZE_T
