Index: openvswitch-2.3.0/datapath/linux/Makefile.main.in
===================================================================
--- openvswitch-2.3.0.orig/datapath/linux/Makefile.main.in
+++ openvswitch-2.3.0/datapath/linux/Makefile.main.in
@@ -71,7 +71,6 @@ default:
 
 modules_install:
 	$(MAKE) -C $(KSRC) M=$(builddir) modules_install
-	depmod `sed -n 's/#define UTS_RELEASE "\([^"]*\)"/\1/p' $(KSRC)/include/generated/utsrelease.h`
 endif
 
 # Much of the kernel build system in this file is derived from Intel's
